### PR TITLE
Don't create package dir when package doesn't exist

### LINKED ISSUE #53: [DNVM.sh] Install missing DNX behaves like the DNX is installed
K Runtime Environment Version Manager - Build 10010

```
kvm list

Active Version              Runtime Architecture Location             Alias
------ -------              ------- ------------ --------             -----
  *    1.0.0-alpha3         mono45  x86          ~/.kre/packages      default
       1.0.0-alpha4-10285   mono45  x86          ~/.kre/packages      
       1.0.0-alpha4-10331   mono45  x86          ~/.kre/packages      
       1.0.0-alpha4-10334   mono45  x86          ~/.kre/packages      

kvm install lateest

Downloading KRE-mono45-x86.lateest from https://www.myget.org/F/aspnetmaster/api/v2
KRE-mono45-x86.lateest was not found in repository https://www.myget.org/F/aspnetmaster/api/v2

kvm list

Active Version              Runtime Architecture Location             Alias
------ -------              ------- ------------ --------             -----
  *    1.0.0-alpha3         mono45  x86          ~/.kre/packages      default
       1.0.0-alpha4-10285   mono45  x86          ~/.kre/packages      
       1.0.0-alpha4-10331   mono45  x86          ~/.kre/packages      
       1.0.0-alpha4-10334   mono45  x86          ~/.kre/packages      
       lateest              mono45  x86          ~/.kre/packages      

mono --version

Mono JIT compiler version 3.8.1 (master/a8a5187 Sun Aug 31 01:02:32 CEST 2014)

cat /etc/issue

Welcome to openSUSE 13.1 "Bottle" - Kernel \r (\l).
```

### PR DESCRIPTION
**Pull request by graemechristie**
Modified kvm_download to download to tmp file and then move to package dir on success.

Fixes #53

### CODE REVIEW

**Comment by Praburaj** [comment]
@graemechristie this PR looks out of date. Will you be able to send an updated PR. We can merge this change. The changes looks good except for one comment that I have.

File: src/kvm.sh [hunk 1]
```diff
@@ -67,6 +67,7 @@ _kvm_download() {
     local pkgName=$(_kvm_package_name "$kreFullName")
     local pkgVersion=$(_kvm_package_version "$kreFullName")
     local url="$KRE_FEED/package/$pkgName/$pkgVersion"
+    local tmpKreFile="$KRE_USER_HOME/$kreFullName.nupkg"
```
Comment (Praburaj) [inline src/kvm.sh]: Can we create the temp in `$KRE_USER_PACKAGES\temp\` like we do on windows?

Comment (Praburaj) [inline src/kvm.sh]: Like this on `kvm.ps` => https://github.com/aspnet/kvm/blob/dev/src/kvm.ps1#L235

**Comment by glennc** [comment]
There is likely no way this will merge cleanly now. Closing, we can redo it if it's still a problem

**Pull request closed without merging**